packages/react-dom/src/FadeIn.tsx

```diff
@@ -0,0 +1,64 @@
+import { type CSSProperties, type ComponentPropsWithoutRef, type ElementType } from 'react'
+import { InView } from './InView'
+import { type InViewOptions } from './useInView'
+import { type OmitKeyof, type Override } from './utility-types'
+
+type FadeInProps<TAs extends ElementType> = Override<
+  ComponentPropsWithoutRef<TAs>,
+  {
+    /**
+     * The element type to render.
+     */
+    as?: TAs
+    /**
+     * The delay in milliseconds before the animation starts.
+     */
+    delay?: number
+    /**
+     * The duration in milliseconds of the animation.
+     */
+    duration?: number
+    /**
+     * The timing function of the animation.
+     */
+    timingFunction?: CSSProperties['animationTimingFunction']
+    /**
+     * The style of the element.
+     */
+    style?: OmitKeyof<CSSProperties, 'opacity' | 'willChange' | 'transition'>
+    /**
+     * The options for the `useInView` hook.
+     */
+    inViewOptions?: InViewOptions
+  }
+>
+
+/**
+ * A component that fades in when it comes into view.
+ */
+export function FadeIn<TAs extends ElementType = 'div'>({
+  as,
+  delay = 0,
+  duration = 200,
```

Comment:
I’m concerned about whether setting the default duration to 200ms is a good idea. The referenced KakaoPay article derived 200ms based on examples from KakaoPay announcements, but I’m not sure if this metric is applicable to us or our users.

- https://tech.kakaopay.com/post/skeleton-ui-idea/

Reply:
I agree your opinion. Do we need to make context provider to set up default prop of FadeIn?

Reply:
> Do we need to make context provider to set up default prop of FadeIn?

Yes, having a provider could help define the scope and manage it consistently!

---

```
"Using repeated animations for any element that takes less than one second to load can be distracting. 
Users may struggle to follow what's happening on the screen and may feel uneasy about the flickering content."
```

This was simply a suggestion to set a default `delay` value.
According to an article by the "UX Research Group Nielsen Norman," animations under one second can be distracting and negatively impact the user experience. Providing a best-practice default value would seem like a good approach.